stock_storage_type/models/stock_storage_category.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2022 ACSONE SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import api, fields, models
+
+
+class StockStorageCategory(models.Model):
+
+    _inherit = "stock.storage.category"
+
+    computed_location_ids = fields.One2many(
+        comodel_name="stock.location", inverse_name="computed_storage_category_id"
+    )
+
+    do_not_mix_lots = fields.Boolean(
```

Comment:
This needs to be replaced by an additional option in the selection `allow_new_product` that you can name `same_lot` = "If all lots are the same"